draft-ietf-tls-esni.md

```diff
@@ -999,7 +999,8 @@ output. Finally, the backend server overwrites the last 8 bytes of the
 ServerHello.random with the following string:
 
 ~~~
-   accept_confirmation = HKDF-Expand-Label(0,
+   accept_confirmation = HKDF-Expand-Label(
+      HKDF-Extract(0, ClientHelloInner.random),
```

Comment:
Do we need to worry about `transcript_ech_conf` itself containing `ClientHelloInner.random` (and a hash of it), or does that work out in the analysis?

Reply:
Hmm... what do you mean by "worry" here?

Reply:
Dunno if there's a problem. This was meant to be a non-leading question as to whether or not it's fine. :-)

My very vague understanding was that formal analyses tended to break down when to you do things like encrypt a key with itself. We're kinda doing that here. So the question is whether this would be a problem.

Reply:
Ah, okay, I see. I sent Hugo a message out of band to get confirmation on the best outcome here. I'll follow up when I hear back. :-)

Reply:
> My very vague understanding was that formal analyses tended to break down when to you do things like encrypt a key with itself. We're kinda doing that here. So the question is whether this would be a problem.

This is a problem for symmetric encryption, but what we're doing here is different enough that I don't think we can draw any conclusions from that. Though looking at Definition 7 in [the HKDF paper](https://eprint.iacr.org/2010/264), it doesn't seem like this has been explicitly considered. My hunch is that this definition can be extended in a way that let's the info string incorporate the IKM, and that the current proof could be modified to handle this.

Reply:
Confirmed with Hugo that this change is not needed if we model the hash function as a RO. If we don't want to do that, and instead want to rely on a PRF assumption, then we'd need to take this suggestion. The existing TLS security proofs already rely on RO liberally, so I think we can safely assume RO here. But I won't block this, either, if folks would be more comfortable with a weaker assumption at the cost of another chain of hashes.

Reply:
The analysis here is in the standard model (no random oracles): https://eprint.iacr.org/2020/1044.pdf

Reply:
My vote would be to do the conservative thing here.

Reply:
I know nothing about this and will defer to you all.

Reply:
It occurred to me that [QUIC _does_ Extract before calling Expand](https://www.rfc-editor.org/rfc/rfc9001.html#name-initial-secrets) in a similar way, so, if for no other reason than consistency, I'd be OK with this change. We might even consider doing something similar with a salt.

Reply:
The signal is already bound to the ECH version via the codepoint, though I suppose we may end up choosing the codepoint in a different way eventually. I'd be OK with that change.